### PR TITLE
Model properties

### PR DESCRIPTION
**Pull request by Faryshta**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes (and new tests added)
| Fixed issues  | #149

### CODE REVIEW

File: tests/ModelGeneratorTest.php [hunk 1]
```diff
@@ -52,50 +53,70 @@ public function relationsProvider()
...
                     'relation' => "\$this->hasMany(Product::className(), ['supplier_id' => 'id']);",
                     'expected' => true,
                 ],
                 [
+                    'name' => 'function getAttributes0()',
+                    'relation' => "\$this->hasMany(Attribute::className(), ['supplier_id' => 'id']);",
+                    'expected' => true,
+                ],
+                [
+                    'name' => 'function getAttributes()',
```
Comment (Faryshta) [inline tests/ModelGeneratorTest.php]: This is the important line, it checks `function getAttributes()` is not created while above it checks `function getAttributes0()` is created.

**Comment by samdark** [comment]
Looks fine to me. @Faryshta could you add a line for changelog?

**Comment by Faryshta** [comment]
@samdark done

**Comment by samdark** [comment]
Merged. Thanks!

**Comment by AnatolyRugalev** [comment]
This commit breaks generation of models in my case:
```
Exception 'yii\base\InvalidConfigException' with message 'The table does not exist: {{%active_record}}'
```

This happens because of my code in my own `common\db\ActiveRecord` class:

```php
    public function behaviors()
    {
        return ArrayHelper::merge(parent::behaviors(), [
            'timestamp' => [
                'class' => TimestampBehavior::className(),
                'createdAtAttribute' => $this->hasAttribute('createdAt') ? 'createdAt' : null,
                'updatedAtAttribute' => $this->hasAttribute('updatedAt') ? 'updatedAt' : null,
                'value' => [$this, 'getTimestampDefaultValue'], // Hack for serialization
            ],
        ]);
    }
```

Generator creates base model class and `$this->trigger(self::EVENT_INIT);` in `BaseActiveRecord:21` produces an `$this->ensureBehaviors()`, which fails because table does not exists. The only workaround I see is to check `static::className()` inside of `behaviors()` which makes me crazy.

Will this be fixed? Or I should find another place for my base behaviors initialization?

**Comment by AnatolyRugalev** [comment]
Base model class is now creating at https://github.com/yiisoft/yii2-gii/pull/148/files#diff-395f09d34b21668a80e7c68456890857R632

**Comment by cebe** [comment]
please open a new issue about it, a merged pull request can not be reopened.

**Pull request merged**